src/mockServiceWorker.js

```diff
@@ -147,20 +147,22 @@ async function handleRequest(event, requestId) {
   if (client && activeClientIds.has(client.id)) {
     ;(async function () {
       const clonedResponse = response.clone()
-      sendToClient(client, {
+      const body =
+        clonedResponse.body === null ? null : await clonedResponse.arrayBuffer()
```

Comment:
Do we have to make this `null` check anymore? I think it was needed because we didn't want to send an empty string if the request didn't have a body, so we could distinguish between explicit "no body" scenario and explicit "empty string" body scenarios on the client-side.

Now, I think empty request body will be read as `ArrayBuffer(0)`, which is totally what we expect on the client, as it will be the developer's responsibility to read that body into something (like `req.text()`, getting an empty string on such an empty buffer). 

The only downside of this is that you won't be able to read explicit `null` with the deprecated `req.body`, as for the library `''` and `null` will both be received as empty buffers, and we wouldn't be able to distinguish when `req.body` reference should return an empty string or a `null`. 

What do you think about this use case?

Reply:
Response constructor throws error if we provide body with 204.